src/auth_proxy.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Observer module for Jenkins to auth_proxy integration."""
+
+import logging
+from typing import List
+
+import ops
+from charms.oathkeeper.v0.auth_proxy import AuthProxyConfig, AuthProxyRequirer
+from charms.traefik_k8s.v2.ingress import IngressPerAppRequirer
+
+import jenkins
+import state
+
+AUTH_PROXY_ALLOWED_ENDPOINTS: List[str] = []
+AUTH_PROXY_HEADERS = ["X-User"]
+
+
+logger = logging.getLogger(__name__)
+
+
+class Observer(ops.Object):
+    """The Jenkins Auth Proxy integration observer."""
+
+    def __init__(self, charm: ops.CharmBase, ingress: IngressPerAppRequirer):
+        """Initialize the observer and register event handlers.
+
+        Args:
+            charm: the parent charm to attach the observer to.
+            ingress: the ingress object from which to extract the necessary settings.
+        """
+        super().__init__(charm, "auth-proxy-observer")
+        self.charm = charm
+        self.ingress = ingress
+
+        self.auth_proxy = AuthProxyRequirer(self.charm)
+
+        self.charm.framework.observe(
+            self.charm.on["auth-proxy"].relation_joined, self._auth_proxy_relation_joined
```

Comment:
A nitpick on this and below, I think the naming convention is `_on_<...>` for event handling functions?

Reply:
Indeed. Updated